src/transform/src/column_knowledge.rs

```diff
@@ -141,17 +141,20 @@ impl ColumnKnowledge {
                     // 2. No fixpoint was found after `max_iterations`. If this
                     //    is the case reset the knowledge vectors for all
                     //    recursive CTEs to DatumKnowledge::top().
-                    // 3. We reach the user-specified iteration limit of any of the bindings.
+                    // 3. We reach the user-specified recursion limit of any of the bindings.
                     //    In this case, we also give up similarly to 2., because we don't want to
                     //    complicate things with handling different limits per binding.
-                    let min_max_iter = max_iters.into_iter().filter_map(|md| *md).min();
+                    let min_max_iter = limits
```

Comment:
nit: we are using this `min_max` calculation at least 2 times and I'm adding one more in my LIR API PR (#19588). Maybe we can have a dedicated ``Vec<Option<LetRecLimit>> ⇒ bool` function  next to the `LetRecLimit` definition.

Reply:
Good idea, done.